File: SGE/ListadoPersonasYDepartamentos/Domain/Interfaces/IUseCaseDepartamentos.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces
{
    public interface IUseCaseDepartamentos
    {
        List<Departamento> GetDepartamentos();
        Departamento GetDetalleDepartamento(int id);
        Departamento GetDepartamentoParaEditar(int id);
        List<Persona> GetPersonasPorDepartamento(int id);
        void CrearDepartamento(Departamento departamento);
        void ActualizarDepartamento(Departamento departamento);
        void EliminarDepartamento(int id);
    }
}

File: Select-persona-Azure/app/domain/interfaces/use-cases/IUseCaseDepartamentos.ts
import { Departamento } from '../../entities/Departamento';
import { Persona } from '../../entities/Persona';

export interface IUseCaseDepartamentos {
  getDepartamentos(): Promise<Departamento[]>;
  getDetalleDepartamento(id: number): Promise<Departamento | null>;
  getPersonasPorDepartamento(id: number): Promise<Persona[]>;
  crearDepartamento(departamento: Departamento): Promise<void>;
  actualizarDepartamento(departamento: Departamento): Promise<void>;
  eliminarDepartamento(id: number): Promise<void>;
}